ee/query-service/app/api/license.go

```diff
@@ -88,6 +88,62 @@ func (ah *APIHandler) applyLicense(w http.ResponseWriter, r *http.Request) {
 	ah.Respond(w, license)
 }
 
+func (ah *APIHandler) listLicensesV3(w http.ResponseWriter, r *http.Request) {
+	licenses, apiError := ah.LM().GetLicensesV3(r.Context())
+
+	if apiError != nil {
+		RespondError(w, apiError, nil)
+		return
+	}
+
+	ah.Respond(w, licenses)
+}
+
+// this function is called by zeus when inserting licenses in the query-service
+func (ah *APIHandler) applyLicenseV3(w http.ResponseWriter, r *http.Request) {
+	var license model.LicenseV3
+
+	if err := json.NewDecoder(r.Body).Decode(&license); err != nil {
+		RespondError(w, model.BadRequest(err), nil)
+		return
+	}
+
+	if license.Key == "" {
+		RespondError(w, model.BadRequest(fmt.Errorf("license key is required")), nil)
+		return
+	}
+
+	l, apiError := ah.LM().ActivateV3(r.Context(), &license)
+	if apiError != nil {
+		RespondError(w, apiError, nil)
+		return
+	}
+
+	ah.Respond(w, l)
+}
+
+func (ah *APIHandler) refreshLicensesV3(w http.ResponseWriter, r *http.Request) {
```

Comment:
In the `refreshLicenses` call, you will only get the license key. Nothing else will be given.

Reply:
done

Reply:
do we need a license key here ? what if the key here doesn't correspond to the active license key currently ? then do set that as active? Since by design each deployment will have only once license and query service is aware of the active license we do not specifically need the license key here

Reply:
This makes sense. You should ideally get nothing and the active license should get refreshed.